ldi-core/ldes-client-v2/tree-node-fetcher/src/main/java/ldes/client/treenodefetcher/TreeNodeFetcher.java

```diff
@@ -0,0 +1,47 @@
+package ldes.client.treenodefetcher;
+
+import ldes.client.requestexecutor.domain.valueobjects.Response;
+import ldes.client.requestexecutor.executor.RequestExecutor;
+import ldes.client.treenodefetcher.domain.entities.TreeNode;
+import ldes.client.treenodefetcher.domain.valueobjects.ModelResponse;
+import ldes.client.treenodefetcher.domain.valueobjects.MutabilityStatus;
+import ldes.client.treenodefetcher.domain.valueobjects.TreeNodeRequest;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.apache.jena.riot.RDFParser;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class TreeNodeFetcher {
```

Comment:
maybe some javadoc what this does? Without the graph on confluence it can be unclear what the difference is between a TreeNodeFetcher and a TreeNodeSupplier